typescript/radix-sdk/src/modules/tx.ts

```diff
@@ -0,0 +1,519 @@
+    gas,
+  }: {
+    token: string;
+    receiver_domain: number;
+    receiver_address: string;
+    gas: string;
+  }) {
+    const transactionManifest = await this.populate.enrollRemoteRouter({
+      from_address: this.account.address,
+      token,
+      receiver_domain,
+      receiver_address,
+      gas,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async unrollRemoteRouter({
+    token,
+    receiver_domain,
+  }: {
+    token: string;
+    receiver_domain: number;
+  }) {
+    const transactionManifest = await this.populate.unrollRemoteRouter({
+      from_address: this.account.address,
+      token,
+      receiver_domain,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async remoteTransfer({
+    token,
+    destination_domain,
+    recipient,
+    amount,
+    custom_hook_id,
+    gas_limit,
+    custom_hook_metadata,
+    max_fee,
+  }: {
+    token: string;
+    destination_domain: number;
+    recipient: string;
+    amount: string;
+    custom_hook_id: string;
+    gas_limit: string;
+    custom_hook_metadata: string;
+    max_fee: { denom: string; amount: string };
+  }) {
+    const transactionManifest = await this.populate.remoteTransfer({
+      from_address: this.account.address,
+      token,
+      destination_domain,
+      recipient,
+      amount,
+      custom_hook_id,
+      gas_limit,
+      custom_hook_metadata,
+      max_fee,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async signAndBroadcast(
+    manifest: TransactionManifest,
+  ): Promise<TransactionHash> {
+    const constructionMetadata =
+      await this.gateway.transaction.innerClient.transactionConstruction();
+
+    const transactionHeader: TransactionHeader = {
+      networkId: this.networkId,
+      startEpochInclusive: constructionMetadata.ledger_state.epoch,
+      endEpochExclusive: constructionMetadata.ledger_state.epoch + 2,
+      nonce: generateRandomNonce(),
+      notaryPublicKey: this.account.publicKey,
+      notaryIsSignatory: true,
+      tipPercentage: 0,
+    };
+
+    const transaction: NotarizedTransaction =
+      await TransactionBuilder.new().then((builder) =>
+        builder
+          .header(transactionHeader)
+          .manifest(manifest)
+          .sign(this.signIntent)
+          .notarize(this.notarizeIntent),
+      );
+
+    const compiledNotarizedTransaction =
+      await RadixEngineToolkit.NotarizedTransaction.compile(transaction);
+
+    const intentHashTransactionId =
+      await RadixEngineToolkit.NotarizedTransaction.intentHash(transaction);
+
+    await this.gateway.transaction.innerClient.transactionSubmit({
+      transactionSubmitRequest: {
+        notarized_transaction_hex: Buffer.from(
+          compiledNotarizedTransaction,
+        ).toString('hex'),
+      },
+    });
+    await this.query.pollForCommit(intentHashTransactionId.id);
+
+    return intentHashTransactionId;
```

Comment:
Does the radix sdk expose a way to simulate transactions before submitting them to avoid sending transactions that might fail?

Reply:
yes with the `query.estimateTransactionFee` method we can simulate it and get the estimated gas amount

Reply:
I think it would be a good idea to add it then if it isn't too much work

Reply:
added it